src/librustdoc/lib.rs

```diff
@@ -566,7 +566,7 @@ fn main_options(options: config::Options) -> MainResult {
                 });
                 info!("finished with rustc");
 
-                krate.version = crate_version;
+                cache.crate_version = crate_version;
```

Comment:
Thanks, this found a bug.

`krate.version` previously was used only during `core::run_global_ctxt` because a cache wasn't available; the version was moved into the cache as soon as it was built in `run_format`. Now that a cache is available in `run_global_ctxt`, it can be assigned there directly.

The (pre-existing) bug this found is that `JsonRenderer::after_krate` calls `krate.version.clone()`. The problem is it does this after it was already moved into the cache, so it will always be `None`. I'll fix it shortly by passing the cache to `after_krate`.

Reply:
So are you going to remove `Crate.version` to remove this footgun? Or did I misunderstand :p

Reply:
> `krate.version` previously was used only during `core::run_global_ctxt` because a cache wasn't available; the version was moved into the cache as soon as it was built in `run_format`. Now that a cache is available in `run_global_ctxt`, it can be assigned there directly.

I feel a bit unclear: the current version of `core::run_global_ctxt` does not appear to mention the word `version`; did you mean that it used to use `krate.version`?

Reply:
> So are you going to remove Crate.version to remove this footgun? Or did I misunderstand :p

Yes, https://github.com/rust-lang/rust/pull/82018/commits/09dc427c649bafcaf7f9f4716d1eb1faf27ca5e7 removes it.

> the current version of core::run_global_ctxt does not appear to mention the word version; did you mean that it used to use krate.version?

I'm using `run_global_ctxt` as an umbrella for all the functions it calls. Here's the previous place where it moved it into the cache: https://github.com/rust-lang/rust/pull/82018/files#diff-93d41f91ceb5e711b3ceaf5f86648871772acac49c271ca948eb06d4dfacfbf9L163
and the diff you're commenting on is where it was set originally.